bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/DeconzBaseThingHandler.java

```diff
@@ -86,39 +97,38 @@ public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
             return;
         }
 
-        if (bridgeStatusInfo.getStatus() == ThingStatus.OFFLINE) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
-            unregisterListener();
-            return;
-        }
-
-        if (bridgeStatusInfo.getStatus() != ThingStatus.ONLINE) {
-            return;
-        }
-
-        Bridge bridge = getBridge();
-        if (bridge == null) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
-            return;
-        }
-        DeconzBridgeHandler bridgeHandler = (DeconzBridgeHandler) bridge.getHandler();
-        if (bridgeHandler == null) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
-            return;
-        }
+        if (bridgeStatusInfo.getStatus() == ThingStatus.ONLINE) {
+            // the bridge is ONLINE, we can communicate with the gateway, so we update the connection parameters and
+            // register the listener
+            Bridge bridge = getBridge();
+            if (bridge == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
```

Comment:
Isn't this rather a configuration error, when the user forgets to specify a bridge in textual configuration?

Reply:
A configuration error should prevent the thing from being initialized. But this can also happen if no callback for the bridge is present.

Reply:
Just tested it with OH3. The framework isn't preventing the Thing from being initialized when the bridge is missing, although specified in XML.

Reply:
This is a core-bug. @openhab/core-maintainers, WDYT?

The thing callback.

Reply:
Hm, Yes. I think it is. It is at least worse to open an issue in OHC for discussion.